src/Control/Carrier/FossaApiClient/Internal/FossaAPIV1.hs

```diff
@@ -1148,10 +1151,23 @@ vsiAddFilesToScan :: (Has (Lift IO) sig m, Has Diagnostics sig m) => ApiOpts ->
 vsiAddFilesToScan apiOpts scanID files = fossaReq $ do
   (baseUrl, baseOpts) <- useApiOpts apiOpts
 
-  let body = VSIAddFilesToScanRequestBody files
+  -- This is in the API layer because it's an implementation detail of the API.
+  -- The API expects all file paths to be normalized to forward slashes.
+  let normalized = Map.fromList . normalizeFiles $ Map.toList files
+  let body = VSIAddFilesToScanRequestBody normalized
   _ <- req POST (vsiAddFilesToScanEndpoint baseUrl scanID) (ReqBodyJsonCompat body) ignoreResponse baseOpts
 
   pure ()
+  where
+    normalizeFiles :: [(Path Rel File, Fingerprint.Combined)] -> [(FilePath, Fingerprint.Combined)]
+    normalizeFiles = fmap normalizeFile
+
+    normalizeFile :: (Path Rel File, Fingerprint.Combined) -> (FilePath, Fingerprint.Combined)
+    normalizeFile (path, fp) = (normalizePath path, fp)
+
+    normalizePath :: Path Rel File -> FilePath
```

Comment:
[optional] unit test

Reply:
I don't think this is really feasible for us to unit test since it is inside the API effect (and I don't think we should have it outside). Plus it's extremely simple. I'd rather solve testing by setting up an integration test with VSI scans E2E.

Going to leave it as is since this is marked optional, but if you disagree happy to talk more!

Reply:
yep yea - feel free to ignore anything with [optional] in my pr comments